### PR TITLE
Create pattern object test

### PR DESCRIPTION
**Pull request by scottnath**
@sergesemashko 

I'm submitting this purely as an example of what a promise could be. The promise works in that it has the proper responses...but, index.js doesn't wait for the response.

Any ideas on what to do next?

thanks,
Scott

### CODE REVIEW

File: lib/create-pattern-object.js [hunk 1]
```diff
@@ -73,7 +74,18 @@ function createPatternObject (patternFolder, callback) {
   /**
    * @TODO: put a promise to return here rather than execute callback
    */
-  return dir.readFiles(patternFolder, matchFiles, process, function (err, files) {callback(err, filesTree)});
+
+  dir.readFilesAsync(patternFolder, matchFiles, process)
+    .then(function(resp){
```
Comment (sergesemashko) [inline lib/create-pattern-object.js]: can we return promise to the calling module and get filesTree there, like:
```js

  // create-pattern-object.js
  return dir.readFilesAsync(patternFolder, matchFiles, process);
}
...

// index.js
createPatternObject('../../app/_patterns').then(function (filesTree) { // do something with filesTree here})
```
?

Comment (scottnath) [inline lib/create-pattern-object.js]: hmmm. so the promise would be in index.js? I'll try!

**Comment by scottnath** [comment]
@sergesemashko I think I'm close, but still not getting promises. I did get the callback working...which at least will allow me to temporarily continue. Not idea though.

**Comment by sergesemashko** [comment]
@scottnath, I think callback is fine

**Comment by scottnath** [comment]
@sergesemashko ok. I'll merge this as is if we're good for now.
I do want to figure out promises though before we move forward too much.

**Comment by sergesemashko** [comment]
sounds good, let's just clean commented code before merging :wink:

**Comment by scottnath** [comment]
actually...can some of it stay? the todos are good and the promises are there as placeholders for when we implement that feature (which should be soon I hope)

**Comment by sergesemashko** [comment]
sure, I meant console.logs

**Comment by scottnath** [comment]
ha
fixed

**Comment by sergesemashko** [comment]
:+1:

**Pull request merged**